### DISCUSSION TITLE
Getting KeyError: 'pore.inlets' when trying to run permeability simulation in OpenPNM

### DISCUSSION THREAD

**Discussion by josephinoo** [PMEAL/OpenPNM#1952] [Q&A]
Hello I am a student of computer science, and I like porous media, thank you very much for the work you have done porespy, has been very helpful to the research team, I would like to take the work we are doing with porepsy, find other properties such as permeability, I tried with examples of openPNM, I have been many errors as KeyError: 'pore.inlets', when applying the simulation.
Attached is the code I have been using, if I am doing something wrong, it would be very helpful if you could help me. 

``` python
import porespy as ps 
import openpnm as op
import matplotlib.pyplot as plt
import numpy as np

im = ps.generators.blobs(shape=[200,200,200],porosity=0.4)
print(ps.metrics.porosity(im))

net= ps.networks.snow(im,
                    voxel_size=1,
                    boundary_faces=['left','right','front','back','top','bottom'],
                    marching_cubes_area=False)
pn=op.network.GenericNetwork()
pn.update(net)
water = op.phases.Water(network=pn)
water.add_model(propname='throat.hydraulic_conductance',
                model=op.models.physics.hydraulic_conductance.valvatne_blunt)

water.add_model(propname='throat.hydraulic_conductance',
                model=op.models.physics.hydraulic_conductance.valvatne_blunt)
flow = op.algorithms.StokesFlow(network=pn, phase=water)
flow.set_value_BC(pores=pn.pores('inlets'), values=200000)
flow.set_value_BC(pores=pn.pores('outlets'), values=100000)
flow.run()
mu = np.mean(water['pore.viscosity'])
delta_P = 100000
delta_P = 100000
Q = np.absolute(flow.rate(pores=pn.pores('inlets')))
Lx = np.amax(pn['pore.coords'][:, 0]) - np.amin(pn['pore.coords'][:, 0])
A = Lx*Lx  
K = Q*mu*Lx/(delta_P*A)
print(K)
```
Apparently there is no such thing as 'pore.inlets'.
![image](https://user-images.githubusercontent.com/46098690/125875350-f9226bad-ee17-4f6a-9736-e22fb74bb534.png)

**Comment by jgostick** [comment]
On the following 2 lines, you use the labels 'inlets' and 'outlets', but these labels are not available:

``` python
flow.set_value_BC(pores=pn.pores('inlets'), values=200000)
flow.set_value_BC(pores=pn.pores('outlets'), values=100000)
```

You can try using other labels, which can see by doing ``print(pn)``.  There should be some labels like 'left' and 'right', or 'x_min' and 'x_max', depending on which version of things you are using.  You can learn more about labels [here](https://github.com/PMEAL/OpenPNM/blob/dev/examples/tutorials/using_and_creating_labels.ipynb)
